koku/api/query_handler.py

```diff
@@ -66,13 +66,21 @@ def __init__(self, query_parameters, url_data,
         self.tenant = tenant
         self.default_ordering = default_ordering
         self.kwargs = kwargs
-        self.resolution = None
         self.time_interval = []
-        self.time_scope_units = None
-        self.time_scope_value = None
         self.start_datetime = None
         self.end_datetime = None
 
+        if self.resolution == 'monthly':
+            self.date_to_string = lambda dt: dt.strftime('%Y-%m')
+            self.string_to_date = lambda dt: datetime.datetime.strptime(dt, '%Y-%m').date()
+            self.date_trunc = TruncMonthString
+            self.gen_time_interval = DateHelper().list_months
+        else:
+            self.date_to_string = lambda dt: dt.strftime('%Y-%m-%d')
+            self.string_to_date = lambda dt: datetime.datetime.strptime(dt, '%Y-%m-%d').date()
+            self.date_trunc = TruncDayString
+            self.gen_time_interval = DateHelper().list_days
```

Comment:
This if-else block got moved in the conversion to using properties. I strongly suspect all of these things can be relocated closer to where they're being used, but I didn't want to expand the scope of this change quite that far. So, they can live in the constructor for now. :smile: